openedx/features/content_type_gating/models.py

```diff
@@ -0,0 +1,70 @@
+"""
+Content Type Gating Configuration Models
+"""
+
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from datetime import datetime
+from django.db import models
+from django.utils.translation import ugettext as _
+
+from student.models import CourseEnrollment
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from openedx.core.djangoapps.config_model_utils.models import StackedConfigurationModel
+from openedx.features.course_duration_limits.config import CONTENT_TYPE_GATING_FLAG, WAFFLE_FLAG_NAMESPACE
+
+
+class ContentTypeGatingConfig(StackedConfigurationModel):
+
+    STACKABLE_FIELDS = ('enabled', 'enabled_as_of', 'studio_override_enabled')
+
+    enabled_as_of = models.DateField(default=None, null=True, verbose_name=_('Enabled As Of'), blank=True)
+    studio_override_enabled = models.NullBooleanField(default=None, verbose_name=_('Studio Override Enabled'), blank=True)
+
+    @classmethod
+    def enabled_for_enrollment(cls, enrollment=None, user=None, course_key=None, today=None):
```

Comment:
is this code the same as in the course_duration_limits model?  
Could it be moved to a config model helper method?

Reply:
Yeah, I couldn't decide whether to have a base-class/helper function, or to just accept that we're going to copy/paste this code because it's only incidentally the same (and might drift in the future).